src/pc_port_specs.erl

```diff
@@ -175,7 +181,14 @@ get_port_spec(Config, OsType, {_Arch, Target, Sources, Opts}) ->
             true  -> cxx;
             false -> cc
         end,
-    Target1 = filename:join(ProjectRoot, Target),
+   
+    Target1 =
+        case os:getenv("REBAR_BARE_COMPILER_OUTPUT_DIR", undefined) of
+            undefined ->
+                filename:join(ProjectRoot, Target);
+            Dir -> 
+                filename:join([ProjectRoot, Dir, Target])
```

Comment:
Are you sure this is right? What is `REBAR_BARE_COMPILER_OUTPUT_DIR` set to that it can be relative to the rebar3 dir?

Reply:
This is from an actual compile in a mix project with this patch which triggers this path above specifically : 

```
"From mix REBAR_BARE_COMPILER_OUTPUT_DIR is /Users/starbelly/devel/erlang/bare/_build/dev/lib/enacl"
===> Fetching pc v1.13.0
===> Analyzing applications...
===> Compiling pc
===> Compiling c_src/aead.c
===> Compiling c_src/enacl.c
===> Compiling c_src/enacl_ext.c
===> Compiling c_src/enacl_nif.c
===> Compiling c_src/generichash.c
===> Compiling c_src/hash.c
===> Compiling c_src/kdf.c
===> Compiling c_src/kx.c
===> Compiling c_src/public.c
===> Compiling c_src/pwhash.c
===> Compiling c_src/randombytes.c
===> Compiling c_src/secret.c
===> Compiling c_src/secretstream.c
===> Compiling c_src/sign.c
===> Linking /Users/starbelly/devel/erlang/bare/deps/enacl/priv/enacl_nif.so
===> Analyzing applications...
===> Compiling enacl
Compiling 1 file (.ex)
Generated bare app
```

Without this patch : 

```
===> Linking /Users/starbelly/devel/erlang/bare/deps/enacl/priv/enacl_nif.so
```

Reply:
Actually, @tsloughter the joining of `ProjectRoot` and `Dir` here is pointless, `filename:join/1` will end up going with `Dir`.  

```
 filename:join(["/Users/starbelly/devel/erlang/bare/deps/enacl", "/Users/starbelly/devel/erlang/bare/_build/dev/lib/enacl"]).
"/Users/starbelly/devel/erlang/bare/_build/dev/lib/enacl"
```

But it doesn't matter, it's just an output dir. But yeah, could just join on `Dir` and and `Target` instead. 

I also tested with jiffy for the legacy path : 

```
iex(1)> :jiffy.encode(%{foo: :bar})
"{\"foo\":\"bar\"}"
iex(2)> :enacl.verify()
:ok
iex(3)>
```

Reply:
Well, I say by chance, but it's documented : 

```
Joins a list of filename Components with directory separators. If one of the elements of Components includes an absolute path, such as "/xxx", the preceding elements, if any, are removed from the result.
```